test/unit/spec/signaling/v2/iceconnectionmonitor.js

```diff
@@ -0,0 +1,172 @@
+'use strict';
+
+const assert = require('assert');
+const sinon = require('sinon');
+
+const IceConnectionMonitor = require('../../../../../lib/signaling/v2/iceconnectionmonitor');
+
+describe('IceConnectionMonitor', () => {
+
+  describe('.constructor', () => {
+    var pc;
+    beforeEach( () => {
+      pc = { foo: 1 };
+    });
+
+    it('stores the peerConnection provided', () => {
+      assert.equal(new IceConnectionMonitor(pc)._peerConnection, pc);
+    });
+
+    it('sets the timer to null', () => {
+      assert.equal(new IceConnectionMonitor(pc)._timer, null);
+      assert.equal(new IceConnectionMonitor(pc)._lastActivity, null);
+    });
+
+    it('defaults to 1sec of check period', () => {
+      assert.equal(new IceConnectionMonitor(pc)._activityCheckPeriodMs, 1000);
+    });
+
+    it('defaults to 3 sec of inactivity threshold', () => {
+      assert.equal(new IceConnectionMonitor(pc)._inactivityThresholdMs, 3000);
+    });
+  });
+
+  describe('.start', () => {
+    // eslint-disable-next-line no-undefined
+    ['foo', 45, { foo: 5 }, null].forEach((callback) => {
+      it(`throws if callback is ${typeof callback} `, () => {
+        const pc = { foo: 1 };
+        const monitor = new IceConnectionMonitor(pc);
+        assert.throws(() => {
+          monitor.start(callback);
+        });
+      });
+    });
+
+    it('starts the timer', () => {
```

Comment:
@makarandp0 I think we can test the callback firing here, which is the only public facing behavior. We should not test internal implementation details like starting/stopping the timer.